apps/OpenSpace/ext/launcher/src/sgctedit/windowcontrol.cpp

```diff
@@ -0,0 +1,521 @@
+        _labelQuality = new QLabel();
+        _labelQuality->setText("Quality:");
+        layoutComboQuality->addWidget(_labelQuality);
+        layoutComboQuality->addWidget(_comboQuality);
+        layoutComboQuality->addStretch(1);
+        layoutProjectionGroup->addLayout(layoutComboQuality);
+        QHBoxLayout* layoutFovH = new QHBoxLayout();
+        _labelFovH = new QLabel();
+        _labelFovH->setText("Horizontal FOV:");
+        layoutFovH->addWidget(_labelFovH);
+        layoutFovH->addWidget(_lineFovH);
+        layoutFovH->addStretch(1);
+        QHBoxLayout* layoutFovV = new QHBoxLayout();
+        _labelFovV = new QLabel();
+        _labelFovV->setText("Vertical FOV:");
+        layoutFovV->addWidget(_labelFovV);
+        layoutFovV->addWidget(_lineFovV);
+        layoutFovV->addStretch(1);
+        layoutProjectionGroup->addLayout(layoutFovH);
+        layoutProjectionGroup->addLayout(layoutFovV);
+        QHBoxLayout* layoutHeightOffset = new QHBoxLayout();
+        _labelHeightOffset = new QLabel();
+        _labelHeightOffset->setText("Height Offset:");
+        layoutHeightOffset->addWidget(_labelHeightOffset);
+        layoutHeightOffset->addWidget(_lineHeightOffset);
+        layoutHeightOffset->addStretch(1);
+        layoutProjectionGroup->addLayout(layoutHeightOffset);
+        layoutCheckboxesFull2->addWidget(borderProjectionGroup);
+        layoutCheckboxesFull1->addLayout(layoutCheckboxesFull2);
+        layoutCheckboxesFull1->addStretch(1);
+        layoutWindowCtrl->addLayout(layoutCheckboxesFull1);
+    }
+    layoutWindowCtrl->addStretch(1);
+    _layoutFullWindow->addLayout(layoutWindowCtrl);
+
+    _comboProjection->setCurrentIndex(0);
+    onProjectionChanged(0);
+    _comboQuality->setCurrentIndex(2);
+
+    return _layoutFullWindow;
+}
+
+void WindowControl::showWindowLabel(const bool show) {
+    _labelWinNum->setVisible(show);
+}
+
+void WindowControl::onSizeXChanged(const QString& newText) {
+    std::string x = newText.toStdString();
+    if (!x.empty()) {
+        _windowDims.setWidth(std::stoi(x));
+    }
+    if (_windowChangeCallback) {
+        _windowChangeCallback(_monIndex, _index, _windowDims);
+    }
+}
+
+void WindowControl::onSizeYChanged(const QString& newText) {
+    std::string y = newText.toStdString();
+    if (!y.empty()) {
+        _windowDims.setHeight(std::stoi(y));
+    }
+    if (_windowChangeCallback) {
+        _windowChangeCallback(_monIndex, _index, _windowDims);
+    }
+}
+
+void WindowControl::onOffsetXChanged(const QString& newText) {
+    std::string xOffset = newText.toStdString();
+    float prevWidth = _windowDims.width();
+    try {
+        if (!xOffset.empty()) {
+            _windowDims.setX(std::stoi(xOffset));
+            _windowDims.setWidth(prevWidth);
+        }
+        if (_windowChangeCallback) {
+            _windowChangeCallback(_monIndex, _index, _windowDims);
+        }
+    }
+    catch (...) {
```

Comment:
Only catch the exception thrown by the validator here;   the `catch (...)` catches every possible exception and is too wide a net here